Typescript/script.ts

```diff
@@ -0,0 +1,274 @@
+    if (randomvar.innerHTML == "<sup>2</sup>√x") {
+      y.innerText = String(Math.sqrt(Number(y.innerText)));
+    } else {
+      y.innerText = String(Math.cbrt(Number(y.innerText)));
+    }
+  }
+}
+function tanpow() {
+  if (y.innerText !== "0" && y.innerText !== "") {
+    if (randompow.innerHTML == "10<sup>x</sup>") {
+      y.innerText = String(Math.pow(10, parseInt(y.innerText)));
+    } else {
+      y.innerText = String(Math.pow(2, parseInt(y.innerText)));
+    }
+  }
+}
+function log() {
+  if (y.innerText !== "0" && y.innerText !== "") {
+    y.innerText = String(Math.log10(Number(y.innerText)));
+  }
+}
+function lnlog() {
+  if (y.innerText !== "0" && y.innerText !== "") {
+    if (randomlog.innerText == "ln") {
+      y.innerText = String(Math.log(Number(y.innerText)));
+    } else {
+      y.innerText = String(Math.E ** Number(y.innerText));
+    }
+  }
+}
+function rational() {
+  if (y.innerText !== "0" && y.innerText !== "") {
+    y.innerText = String(1 / parseInt(y.innerText));
+  }
+}
+const factorial = () => {
+  let fact = 1;
+  let num = Number(y.innerText);
+  console.log(fact);
+  for (let i = 1; i <= num; i++) {
+    fact *= i;
+    console.log(fact);
+  }
+  y.innerText = String(fact);
+}
+function exponentiation() {
+  if (y.innerText !== "0" && y.innerText !== "") {
+    let a: string = (y.innerText += "**");
+    printOutput("");
+    printHistory(a);
+  }
+}
+function scientific_nota() {
+  if (y.innerText !== "0" && y.innerText !== "") {
+    let x: string = y.innerText;
+    let a: string = parseInt(x).toExponential();
+    printOutput("");
+    printHistory(a);
+  }
+}
+
+let mv: number[] = [];
+function modx() {
+  if (y.innerText !== "0" && y.innerText !== "") {
+    y.innerText = String(0 - parseInt(y.innerText));
+  }
+}
+function ms() {
+  if (y.innerText !== "0" && y.innerText !== "") {
+    mv.unshift(Number(y.innerText));
+    let z = document.getElementById("memory-value")! as HTMLInputElement;
+    z.innerText = String(mv);
+  }
+}
+function mp() {
+  if (mv[0]) {
+    y.innerText = String(mv[0] + parseInt(y.innerText));
+  }
+}
+function mm() {
+  if (mv[0]) {
+    y.innerText = String(mv[0] - parseInt(y.innerText));
+  }
+}
+function mc() {
+  let z = document.getElementById("memory-value")! as HTMLInputElement;
+  z.innerText = "";
+  mv = [];
+}
+function mr() {
+  if (y.innerText !== "0" && y.innerText !== "") {
+    y.innerText += mv[0];
+  }
+}
+const deg_rad = () => {
+  let val = document.getElementById("deg")! as HTMLInputElement;
+  let convert: number = parseInt(y.innerText);
+  if (y.innerText !== "0" && y.innerText !== "") {
+    if (val.innerText === "DEG") {
+      val.innerText = "RAD";
+      console.log(val.innerText);
+      y.innerText = String(convert * (180 / Math.PI));
+    } else {
+      val.innerText = "DEG";
+      y.innerText = String(convert * (Math.PI / 180));
+      console.log(val.innerText);
+    }
+  }
+};
+let valid: boolean = true;
```

Comment:
`boolean` type will be automatically inferred since you've initialized the variable with a boolean value.
No need to add type explicitly.